core/src/main/kotlin/entity/User.kt

```diff
@@ -24,7 +23,14 @@ open class User(
     override val id: Snowflake
         get() = data.id
 
-    val avatar: Avatar get() = Avatar(data, kord)
+    /**
+     * The users avatar as [Icon] object
+     */
+    val avatar: Icon
+        get() = when {
+            data.avatar != null -> Icon.UserAvatar(kord, data)
```

Comment:
split those into a defaultAvatar and an avatar one that is nullable instead of squashing both into a this or that fallback logic. and the user then would be able to pick if they want to fall back to the default avatar.

With this. the `DefaultAvatar` sealed instance would become redundant.

Reply:
I would still like to maintain the DefaultAvatar instance in the Icon class, as this is kind of the "Route" enum. Making the avatar nullable and having the user fall back to the "defaultAvatar" is fine for me though